test_model.py

```diff
@@ -28,23 +31,27 @@
     queries_and_output = load_queries_and_output(queries_path, queries_output_path)
 
     if args.model == "boolean":
+        Path("data/boolean_results").mkdir(parents=True, exist_ok=True)
```

Comment:
You can use `save_strings_as_text_file` and remove this logic, which is already done in the function.